torchvision/features/_bounding_box.py

```diff
@@ -0,0 +1,102 @@
+import enum
+from typing import Any, Union
+
+import torch
+
+from ._core import Feature
+
+__all__ = ["BoundingBox", "BoundingBoxFormat"]
+
+
+class BoundingBoxFormat(enum.Enum):
+    XYXY = "XYXY"
+    XYWH = "XYWH"
+    CXCYWH = "CXCYWH"
+
+
+class BoundingBox(Feature):
+    formats = BoundingBoxFormat
+
+    @staticmethod
+    def _parse_format(format: Union[str, BoundingBoxFormat]) -> BoundingBoxFormat:
+        if isinstance(format, str):
+            format = format.upper()
+        return BoundingBox.formats(format)
+
+    def __init__(self, data: Any = None, *, format: Union[str, BoundingBoxFormat], requires_grad=False):
+        super().__init__()
+        self._format = self._parse_format(format)
+        self._to_xyxy = {
+            self.formats.XYWH: self._from_xywh,
+            self.formats.CXCYWH: self._from_cxcywh,
+        }
+        self._from_xyxy = {
+            self.formats.XYWH: self._to_xywh,
+            self.formats.CXCYWH: self._to_cxcywh,
+        }
+
+    def __new__(cls, data: Any = None, *, format: Union[str, BoundingBoxFormat], requires_grad=False):
+        return super().__new__(cls, data, requires_grad=requires_grad)
+
+    @property
+    def format(self) -> BoundingBoxFormat:
+        return self._format
+
+    def convert(self, format: Union[str, BoundingBoxFormat]) -> "BoundingBox":
```

Comment:
Nice! One other thing to think about is if there would be a minimum common API across all our data types that we might want to enforce?

Reply:
Currently, I don't see anything. Happy to hear ideas though. If something comes up later, that should be retro-fittable.

Reply:
I've added a `TensorFeature.from_tensor` method, that should make it easy to create a new feature from just a tensor.